### ISSUE TITLE
Why dose fp8 quantization use multiplication by scale ?

### ISSUE THREAD

**Issue by luliyucoordinate** [NVIDIA/TransformerEngine#477]
Why dose fp8 quantization use multiplication by scale ?

transformer engine quant

https://github.com/NVIDIA/TransformerEngine/blob/2c410836b77667300d4e7d5fa4b87304638fcc1a/transformer_engine/common/util/vectorized_pointwise.h#L215

dequant 

https://github.com/NVIDIA/TransformerEngine/blob/2c410836b77667300d4e7d5fa4b87304638fcc1a/transformer_engine/common/util/cast.cu#L27

pytorch quant

![image](https://github.com/NVIDIA/TransformerEngine/assets/17606906/ba40ad8b-2d1d-48c3-bba2-7cf5d1c5d0c4)
https://pytorch.org/docs/stable/quantization.html

@ptrendx @ksivaman

**Comment by timmoon10** [comment]
FP8 has a limited dynamic range, so the scaling factors are important for avoiding numerical overflows and mitigating numerical underflows. This is especially important for training since gradients tend to have a larger dynamic ranges and the numerical accuracy of gradients can have a significant impact on convergence. You can imagine this as a generalization of FP16 training, which uses a single scaling factor for all gradients. FP8 is more delicate, so it requires per-tensor scaling factors for both the forward and backward passes.

More explanation in the docs: https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/examples/fp8_primer.html

**Comment by luliyucoordinate** [comment]
Most quantization methods divide by scale, such as in Pytorch, while in TE, the multiply by scale. Why?

**Comment by sudhakarsingh27** [comment]
Dequant op multiplies by `scale_inv` (inverse of the `scale`) which is the same as dividing by `scale`.

**Comment by ptrendx** [comment]
I think the question is mostly about naming - ultimately which of the values we call "scale" and "scale inverse" does not really matter as long as we are consistent. I see how it could be confusing for someone coming from a different convention though.

**Comment by luliyucoordinate** [comment]
> Dequant op multiplies by `scale_inv` (inverse of the `scale`) which is the same as dividing by `scale`.

Uh, what confused me is that traditional quantization algorithms use division for the quantization process and multiplication by scale for the de-quantization process. However, in TE, it's exactly the opposite, multiplication is used for the quantization process and multiplication by scale_inv (division by scale) is used for the de-quantization process.
